### PR TITLE
NAS-130979 / 25.04 / Support legacy API

### PR DESCRIPTION
**Pull request by themylogin**
Continue implementing NEP 046 (Versioned API).

This code adds `/api/vXX.XX.X` endpoints for all API versions we have and provides data conversion routines between those versions.

There are no integration tests because we don't yet have two different API versions. The code is extensively covered by unit-tests. The integration testing was performed by adding a fake 24.10.0 API and ensuring everything works smoothly. Fake API code is not present in this PR.

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]
Really cool, please wait for @anodos325 review before merge.

**Review by anodos325 — APPROVED** [review]

**Comment by themylogin** [comment]
time 40:00

**Pull request merged**
